gobblin-modules/gobblin-json/src/main/java/gobblin/writer/SimpleJsonCompressionDataWriter.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.writer;
+
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.io.compress.CompressionCodec;
+import org.apache.hadoop.io.compress.CompressionCodecFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedList;
+
+
+/**
+ * An implementation of {@link DataWriter} that writes json objects directly to HDFS.
+ *
+ * This class accepts one new configuration parameters:
+ * <ul>
+ * <li>{@link SIMPLE_WRITER_COMPRESSION_CODEC} accepts a codec name, which implements the
+ * {@link org.apache.hadoop.io.compress.CompressionCodec} interface like {@link org.apache.hadoop.io.compress.GzipCodec}
+ * </ul>
+ */
+public class SimpleJsonCompressionDataWriter extends FsDataWriter<JsonElement> {
```

Comment:
Just wondering if this can be CompressionDataWriter for generic FsDataWriter<? implements Serializable> rather than being specific to JsonElement?